src/DnsHelperUI/MainForm.cs

```diff
@@ -34,9 +34,23 @@ private void MainForm_Load(object sender, EventArgs e)
             // Load the DNS server config from the file
             ReloadContextMenuItems();
 
-            // Discover the current nic, just naively select the first one 
-            // (this will not work on systems with multiple enabled NIC's)
-            CurrentNic = NetworkManagement.GetAllNicDescriptions().FirstOrDefault();
+            // Get all the nics and load them into a drop down, 
+            // if we have any NICs, fill dropdown and 
+            // select the first one from the list as current.
+            var nics = NetworkManagement.GetAllNicDescriptions();
+            
+            if (nics.Count() > 0)
```

Comment:
Consider using the `.Length` property or the `.Any()` extension method instead of `.Count()`.